cpp/src/arrow/compute/kernels/cast.cc

```diff
@@ -278,34 +278,36 @@ struct CastFunctor<O, I, typename std::enable_if<is_float_truncate<O, I>::value>
     const in_type* in_data = GetValues<in_type>(input, 1);
     auto out_data = GetMutableValues<out_type>(output, 1);
 
-    if (options.allow_float_truncate) {
```

Comment:
should we have a specialized fast path for `options.allow_float_truncate && options.allow_float_overflow`?

Reply:
yes